packages/runtime-core/src/componentProps.ts

```diff
@@ -156,8 +156,10 @@ export function updateProps(
   const [options] = instance.propsOptions
 
   if (
-    // always force full diff if hmr is enabled
-    !(__DEV__ && instance.type.__hmrId) &&
```

Comment:
I think instead we should be checking `instance.parent && instance.parent.type.__hmrId`. Can you confirm that also fixes the issue?

Reply:
Yeah. `instance.parent && instance.parent.type.__hmrId` also fix the issues.